### PR TITLE
Ticket 5 get comments by article

### PR DESCRIPTION
**Pull request by joshnajoshy**
Test checks function responds with 200 and an array of comments that are specific to article_id

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -118,4 +118,50 @@ describe('GET: /api/articles/:article_id', () => {
...
+    })
+
+    describe('GET: /api/articles/:article_id/comments', () => {
+      test('200: responds with an array of comments for the specific article', () => {
+      return request(app)
+      .get('/api/articles/5/comments')
+      .expect(200)
+      .then(({body}) => {
+        const {comments} = body
+        console.log(comments)
```
Comment (nc-alex-davis) [inline __tests__/app.test.js:130]: please remove floaty logs before opening PRs
PR: _production quality code_

File: __tests__/app.test.js [hunk 2]
```diff
@@ -118,4 +118,50 @@ describe('GET: /api/articles/:article_id', () => {
...
+        expect(comments.length).toBe(2) 
+        const result = []
+        comments.filter((comment) => {
+          result.push(comment.created_at)
```
Comment (nc-alex-davis) [inline __tests__/app.test.js:134]: I'm not sure what this is supposed to do, but I'm confused as to 
1. why you're using filter to push to a new array (this is definitely an anti-pattern)

Otherwise, this is a kinda cool workaround for checking the sorting though.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -118,4 +118,50 @@ describe('GET: /api/articles/:article_id', () => {
...
+        })
+        const expectedResult = [...result].sort((a,b) => b - a)
+        expect(result[0]).toEqual(expectedResult[0])
+        comments.forEach((comment) => {
+          expect(comment).toMatchObject({
```
Comment (nc-alex-davis) [inline __tests__/app.test.js:139]: I believe that `toMatchObject` will match with an array of values, too
so, consider having a go at refactoring to pass the whole comments array to the initial expect

🧐

File: __tests__/app.test.js [hunk 4]
```diff
@@ -118,4 +118,50 @@ describe('GET: /api/articles/:article_id', () => {
...
+        .then(({body}) => {
+          expect(body.msg).toBe('bad request');
+        })
+    })
+    test('404: responds with comments not found if article_id is a number but article not found', () => {
+      return request(app)
+      .get('/api/articles/1000/comments')
+      .expect(404)
+      .then(({body}) => {
+        expect(body.msg).toBe('comments not found');
```
Comment (nc-alex-davis) [inline __tests__/app.test.js:164]: this message should probably be article not found, no? Thinking in terms of resources as communicated by the endpoint string, _comments not found_ implies the article was found ok, but its comments weren't

File: controllers/api.controllers.js [hunk 5]
```diff
@@ -20,10 +20,19 @@ fetchArticleById(article_id).then((article) => {
...
 
-module.exports = {getAllEndpoints, getAllTopics, getArticlesById, getAllArticles};
\ No newline at end of file
+const getAllCommentsByArticleId = (request, response, next) => {
+    const {article_id} = request.params
+    fetchAllCommentsByArticleId(article_id).then((comments) => {
+        response.status(200).send({comments})
+    }).catch((error) => {
+        next(error)
+    })
```
Comment (nc-alex-davis) [inline controllers/api.controllers.js:35]: good 'n' clean 😄

File: endpoints.json [hunk 6]
```diff
@@ -41,5 +41,29 @@
...
+        {
+          "comment_id": 14,
+          "article_id": 5,
+          "body": "What do you see? I have no idea where this will lead us. This place I speak of, is known as the Black Lodge.",
+          "votes": 16,
+          "author": "icellusedkars",
+          "created_at": "2020-06-09T05:00:00.000Z"
+        }
+      ]
+    }
```
Comment (nc-alex-davis) [inline endpoints.json:67]: clearly communicated 👍

File: models/api.models.js [hunk 7]
```diff
@@ -27,4 +27,13 @@ const fetchAllArticles = () => {
     })
 }
 
-module.exports = {fetchAllTopics, fetchArticleById, fetchAllArticles};
\ No newline at end of file
+const fetchAllCommentsByArticleId = (article_id) => {
+return db.query(`SELECT * FROM comments WHERE article_id = $1 ORDER BY created_at DESC`, [article_id]).then(({rows}) => {
+    if(rows.length === 0){
+        return Promise.reject({status: 404, msg: 'comments not found'})
```
Comment (nc-alex-davis) [inline models/api.models.js:33]: you might benefit from some kind of `checkExists` logic here, to determine whether it's the article that's missing, or whether it has no comments yet

because it's fine to return an empty array if the article exists but has no comments yet. That's 200 OK

File: __tests__/app.test.js (hunk 4 above)
Comment (nc-alex-davis) [reply __tests__/app.test.js:164]: You also should test for 200 OK and empty array if the article is found but has no comments yet

**Review by nc-alex-davis — COMMENTED** [review]
Overall this is decent stuff – well done

There are some rather important adjustments to be made regarding empty arrays, article not found vs comments not found
